File: 2244.minimum-rounds-to-complete-all-tasks/solution.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:

        # Counts up each task id
        counts = Counter(tasks)

        # Calculates the number of rounds
        result = 0
        for taskId in counts:
            cnt = counts[taskId]
            if cnt == 1:
                return -1

            # Since we must do 2 or 3 tasks each round,
            # calculates both cases, and then pick the minimum number
            div2 = int(cnt / 2) + (cnt % 2)
            div3 = int(cnt / 3) + (1 if (cnt % 3 > 0) else 0)
            result += min(div2, div3)

        return result

File: 2244.minimum-rounds-to-complete-all-tasks/solution.kt
class Solution {
    fun minimumRounds(tasks: IntArray): Int {
        val counts = HashMap<Int, Int>()
        var result = 0

        // Counts up each taskId
        for (taskId in tasks) {
            counts.put(taskId, (counts.get(taskId) ?: 0) + 1)
        }

        // Calculates the rounds
        for (taskId in counts.keys) {
            val taskCnt = counts.get(taskId) ?: 1
            if (taskCnt == 1) return -1

            val div2 = (taskCnt / 2) + (taskCnt % 2)
            val div3 = (taskCnt / 3) + when(taskCnt % 3) {
                0 -> 0
                else -> 1
            }
            result += minOf(div2, div3)
        }

        return result
    }
}
